crates/vm/levm/src/vm.rs

```diff
@@ -200,11 +200,202 @@ pub fn word_to_address(word: U256) -> Address {
     Address::from_slice(&bytes[12..])
 }
 
+#[allow(clippy::too_many_arguments)]
+fn create_transaction(
+    to: Address,
+    msg_sender: Address,
+    value: U256,
+    calldata: Bytes,
+    gas_limit: U256,
+    block_number: U256,
+    coinbase: Address,
+    timestamp: U256,
+    prev_randao: Option<H256>,
+    chain_id: U256,
+    base_fee_per_gas: U256,
+    gas_price: U256,
+    db: Db,
+    block_blob_gas_used: Option<U256>,
+    block_excess_blob_gas: Option<U256>,
+    tx_blob_hashes: Option<Vec<H256>>,
+) -> Result<VM, VMError> {
+    let bytecode = db.get_account_bytecode(&to);
+
+    let initial_call_frame = CallFrame::new(
+        msg_sender,
+        to,
+        to,
+        None,
+        bytecode,
+        value,
+        calldata.clone(),
+        false,
+        gas_limit,
+        TX_BASE_COST,
+        0,
+    );
+
+    let env = Environment {
+        consumed_gas: TX_BASE_COST,
+        origin: msg_sender,
+        refunded_gas: U256::zero(),
+        gas_limit,
+        block_number,
+        coinbase,
+        timestamp,
+        prev_randao,
+        chain_id,
+        base_fee_per_gas,
+        gas_price,
+        block_blob_gas_used,
+        block_excess_blob_gas,
+        tx_blob_hashes,
+    };
+
+    Ok(VM {
+        call_frames: vec![initial_call_frame],
+        db,
+        env,
+        accrued_substate: Substate::default(),
+    })
+}
+
+#[allow(clippy::too_many_arguments)]
+fn create_contract(
+    sender: Address,
+    secret_key: H256,
+    db: &mut Db,
+    value: U256,
+    calldata: Bytes,
+    block_number: U256,
+    coinbase: Address,
+    timestamp: U256,
+    prev_randao: Option<H256>,
+    chain_id: U256,
+    base_fee_per_gas: U256,
+    gas_price: U256,
+    block_blob_gas_used: Option<U256>,
+    block_excess_blob_gas: Option<U256>,
+    tx_blob_hashes: Option<Vec<H256>>,
+    salt: Option<U256>,
+) -> Result<VM, VMError> {
+    /*
+    Functionality should be:
+    (1) Check whether caller has enough balance to make a transfer
+    (2) Derive the new contract’s address from the caller’s address (passing in the creator account’s nonce)
+    (3) Create the new contract account using the derived contract address (changing the “world state” StateDB)
+    (4) Transfer the initial Ether endowment from caller to the new contract
+    (5) Set input data as contract’s deploy code, then execute it with EVM. The ret variable is the returned contract code
+    (6) Check for error. Or if the contract code is too big, fail. Charge the user gas then set the contract code
+    Source: https://medium.com/@hayeah/diving-into-the-ethereum-vm-part-5-the-smart-contract-creation-process-cb7b6133b855
+     */
+    let mut db_copy = db.clone();
+    let mut sender_account = match db_copy.accounts.get(&sender) {
+        Some(acc) => acc,
+        None => {
+            return Err(VMError::SenderAccountDoesNotExist);
```

Comment:
Although we use a database to track account information, if an account is not in our database, it doesn't mean it doesn't exist but that it has no balance. Accounts are not created in Ethereum; all accounts exist; the only difference is whether they have a balance or not.
```suggestion
            return Err(VMError::OutOfGas);
```
As an exercise, sample a random account and try to send any transaction, or ask its balance.